src/metabase/api/card.clj

```diff
@@ -158,6 +148,23 @@
                      card)))
             cards))))
 
+(defn hydrate-for-frontend
```

Comment:
I extracted the common parts out of the endpoints that return a card: `GET, POST, and PUT /api/card`, and also `POST /api/card/from-csv`